SpatialGDK/Source/SpatialGDK/Private/Tests/SpatialView/TargetView.cpp

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "Tests/SpatialView/TargetView.h"
+
+#include "SpatialView/EntityComponentTypes.h"
+
+namespace SpatialGDK
+{
+void FTargetView::AddEntity(Worker_EntityId EntityId)
+{
+	Builder.AddEntity(EntityId);
+}
+
+void FTargetView::RemoveEntity(Worker_EntityId EntityId)
+{
+	// Make sure ops are generated to remove all authority and components for the removed entity.
+	EntityViewElement EntityData = View.FindAndRemoveChecked(EntityId);
+	for (const Worker_ComponentSetId& SetId : EntityData.Authority)
+	{
+		Builder.SetAuthority(EntityId, SetId, WORKER_AUTHORITY_NOT_AUTHORITATIVE, {});
+	}
+	for (const ComponentData& Component : EntityData.Components)
+	{
+		Builder.RemoveComponent(EntityId, Component.GetComponentId());
+	}
+	Builder.RemoveEntity(EntityId);
+}
+
+void FTargetView::AddOrSetComponent(Worker_EntityId EntityId, ComponentData Data)
+{
+	EntityViewElement* EntityData = View.Find(EntityId);
```

Comment:
FindOrAdd?

Reply:
I don't like findoradd because it's not finding and returning something, it's setting a value. It's kind of like [] in std::unordered_map

Reply:
The action of this is - if this component doesn't exist then add it. If it does it then set it's value to this.